File: tetrahedron.cpp
//------------------------------------------------------------------------------
// tetrahedron_In.cpp - содержит процедуру ввода параметров
// для уже созданного тетраэдра
//------------------------------------------------------------------------------

#include "tetrahedron.h"
#include "math.h"

//------------------------------------------------------------------------------
// Ввод параметров тетраэдра из файла
void In(tetrahedron &t, ifstream &ifst) {
    ifst >> t.a;
}

// Случайный ввод параметров тетраэдра
void InRnd(tetrahedron &t) {
    t.a = Random();
    t.density = RandomDouble();
}

//------------------------------------------------------------------------------
// Вывод параметров тетраэдра в форматируемый поток
void Out(tetrahedron &t, ofstream &ofst) {
    ofst << "It is Tetrahedron: a = " << t.a
         << ". SurfaceArea = " << SurfaceArea(t) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление периметра тетраэдра
double SurfaceArea(tetrahedron &t) {
    return std::sqrt(3) * t.a * t.a;
}

File: avs3/tetrahedron.py
from math import sqrt
import random


# ----------------------------------------------
# Ввод параметров тетраэдра из файла.
def in_file(tetrahedron, line):
    str_array = line.split(' ')
    tetrahedron.append("tetrahedron")
    tetrahedron.append(float(str_array[1]))  # density.
    tetrahedron.append(int(str_array[2]))  # a.
    tetrahedron.append(surface_area(tetrahedron))  # Surface area.


# Случайный ввод параметров тетраэдра.
def in_rnd(tetrahedron):
    tetrahedron.append("tetrahedron")
    tetrahedron.append(round(random.uniform(0.1, 20.0), 3))  # density.
    tetrahedron.append(random.randint(1, 20))  # a.
    tetrahedron.append(surface_area(tetrahedron))  # Surface area.


# Вычисление плозади поверхности тетраэдра.
def surface_area(tetrahedron):
    return float(sqrt(3) * tetrahedron[2] * tetrahedron[2])
